src/esp/assets/ResourceManager.h

```diff
@@ -487,13 +487,48 @@ class ResourceManager {
   void translateMesh(BaseMesh* meshDataGL, Magnum::Vector3 translation);
 
   /**
-   * @brief Compute and return the axis aligned bounding box of a mesh.
-   *
+   * @brief Compute and return the axis aligned bounding box of a mesh in mesh
+   * local space
    * @param meshDataGL The mesh data.
    * @return The mesh bounding box.
    */
   Magnum::Range3D computeMeshBB(BaseMesh* meshDataGL);
 
+  /**
+   * @brief Compute the absolute AABBs for drawables in PTex mesh in world space
+   * @param baseMesh: ptex mesh
+   */
+
+  void computePTexMeshAbsoluteAABBs(BaseMesh& baseMesh);
+
+  /**
+   * @brief Compute the absolute AABBs for drawables in general mesh (e.g.,
+   * MP3D) world space
+   */
+  void computeGeneralMeshAbsoluteAABBs();
+
+  /**
+   * @brief Compute absolute transformations of all drwables stored in
+   * staticDrawableInfo_
+   */
+  std::vector<Magnum::Matrix4> computeAbsoluteTransformations();
+
+  /*
+   * @brief this helper vector contains information of the drawables on which
+   * we will compute the absolute AABB pair: <Drawable's scene node, meshID>
+   *
+   * -) non-ptex mesh:
+   * meshID is the global index into meshes_.
+   * meshes_[meshID] is the BaseMesh corresponding to the drawable;
+   *
+   * -) ptex mesh:
+   * meshID is the index of the submesh corresponding to the drawable;
+   */
+  std::vector<
```

Comment:
Maybe use a proper struct instead of a pair since this is used in such a large scope.

It also seems like this should really be a local variable instead of a member for this.